✨Feat 프로필 조회, 수정, 내가 쓴 글, 찜목록, 회원 탈퇴 구현

hyunzoo123123 opened the pull request:
## 의도
어떤 의도로 코드를 작성했는지 적어주세요
- 프로필 조회, 수정, 내가 쓴 글, 찜목록, 회원 탈퇴

## 주요 변경점
어떤 부분이 변경되었는지 적어주세요
- 회원 탈퇴 시 내가 작성한 게시글, 댓글 모두 삭제 되는 기능

## ETC
리뷰어들에게 어떤 점을 검토해야하는지 알려주세요
- 꼼꼼하게 봐주세요

HwangSeungHyeon committed:
✨Feat: 목록 조회 기능 구현

MemberProfileUpdateRequest: 수정 기능 생성
MemberController: 수정,조회, 찜목록, 내가 쓴 글 목록 조회 생성
MemberService: 수정,조회, 찜목록, 내가 쓴 글 목록 조회 생성
MemberServiceImpl: 수정,조회, 찜목록, 내가 쓴 글 목록 조회 생성

Resolves: #12

hyunzoo123123 committed:
✨Feat: 목록 조회 기능 구현

SecurityConfig: "/members/**",  추가

Resolves: #12

hyunzoo123123 committed:
✨Feat: 목록 조회 기능 구현

SecurityConfig: "/members/**"  추가

Resolves: #12

hyunzoo123123 committed:
✨Feat: 목록 조회 기능 구현

MemberServiceImpl: 의미 없는 공백 제거

Resolves: #12

hyunzoo123123 committed:
✨Feat: 목록 조회 기능 구현

Resolves: #12

hyunzoo123123 committed:
✨Feat: 회원 탈퇴 구현 중 (임시)

Resolves: #12

hyunzoo123123 committed:
✨Feat: 회원 탈퇴 구현 중 (임시)

소프트딜리트 기능 구현 중

Resolves: #12

hyunzoo123123 committed:
Merge branch 'dev' of https://github.com/HwangSeungHyeon/buysell into feature/profile

# Conflicts:
#	src/main/kotlin/com/teamsparta/buysell/domain/member/model/Member.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/post/model/Post.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/post/repository/PostRepositoryImpl.kt
#	src/main/kotlin/com/teamsparta/buysell/domain/post/service/PostServiceImpl.kt

hyunzoo123123 committed:
🔨Refactor: 회원 탈퇴 구현 중

MemberServiceImpl 코드 수정

Resolves: #12

hyunzoo123123 committed:
🔨Refactor: 회원 탈퇴

Member, MemberProfileUpdateRequest,updateMember: gender 제거

Resolves: #12

hyunzoo123123 committed:
🔨Refactor: 회원 탈퇴

Member, MemberProfileUpdateRequest,updateMember: gender 제거

Resolves: #12

lazzzykim commented on src/main/resources/application.yml:
```diff
@@ -1,13 +1,12 @@
-    password: ${postgresql.password}
+    url: jdbc:postgresql://aws-0-ap-northeast-2.pooler.supabase.com:5432/postgres
+    username: postgres.phpemzzszbjmdwprloda
+    password: D98%,?YA/!GYc(r
```
연결 정보가 노출되는 것은 위험하다고 생각됩니다
환경변수에서 설정하는것이 좋아보입니다!

lazzzykim commented on src/main/kotlin/com/teamsparta/buysell/domain/member/controller/MemberController.kt:
```diff
@@ -23,11 +28,66 @@ class MemberController(
+    }
+
+    @PutMapping("/signout")
+    fun Pretenddelete(
```
함수명 Pretenddelete 에서 카멜케이스로 변경해주세요~~

hyunzoo123123 committed:
🔨Refactor: 회원 탈퇴

MemberService, MemberController, MemberServiceImpl: Pretenddelete > pretendDelete 수정

MemberStatus:Pretenddelete > PRETENDDELETE 수정

Resolves: #12

hyunzoo123123 committed:
🔨Refactor: 회원 탈퇴

yml 수정

Resolves: #12

hyunzoo123123 committed:
🔨Refactor: 회원 탈퇴

Member: @Where 제거
MemberController: Pretenddelete >  pretendDelete

Resolves: #12

codekmj1 reviewed (approved):
확인했습니다

HwangSeungHyeon commented on src/main/kotlin/com/teamsparta/buysell/domain/member/controller/MemberController.kt:
```diff
@@ -23,11 +28,66 @@ class MemberController(
+        memberService.pretendDelete(userPrincipal)
+        return ResponseEntity
+            .status(HttpStatus.OK)
+            .body("탈퇴 신청 완료!")
```
나중에 저희 개발 다하고 나면, 문자를 바디에 담을지, messege를 전달하는 DTO를 이용할지 결정하면 좋을것같습니다.

HwangSeungHyeon committed:
Merge branch 'dev' into feature/profile

lazzzykim reviewed (approved):
고생하셨습니다

HwangSeungHyeon reviewed (approved):
어려운 내용 구현하시느라 고생많으셨습니다.
